sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -196,7 +198,8 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
                 hof.dataTypeMismatch(hof, checkRes)
               case TypeCheckResult.TypeCheckFailure(message) =>
```

Comment:
Will it still match `TypeCheckResult.TypeCheckFailure(message)`?

Reply:
When we finish all tasks of converting the type check failures to error classes, we will remove `TypeCheckResult.TypeCheckFailure()`, and as a consequence of that, we can remove all match cases, and unneeded error classes like this. 

So far, there are open tasks, I wouldn't do that:
- https://issues.apache.org/jira/browse/SPARK-40371
- https://issues.apache.org/jira/browse/SPARK-40372
- https://issues.apache.org/jira/browse/SPARK-40374
- https://issues.apache.org/jira/browse/SPARK-40748
- https://issues.apache.org/jira/browse/SPARK-40749
- https://issues.apache.org/jira/browse/SPARK-40769

since there is a risk to miss something.